.github/workflows/ci.yml

```
@@ -1,18 +1,46 @@
-  build:
+  detect-changes:
+    name: 🔍 Detect Changed Modules
     runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.filter.outputs.changes }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: 📁 Detect changes in service modules
+        id: filter
+        uses: dorny/paths-filter@v2
```

actionlint (1.7.4)

the runner of "dorny/paths-filter@v2" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)